src/pool/srv_layout.h

```diff
@@ -78,4 +79,10 @@ struct pool_hdl {
 
 extern daos_prop_t pool_prop_default;
 
+/**
+ * Performs lazy initialization of any pool props that need to be dynamically
+ * allocated.
+ */
+void lazy_init_default_pool_props(void);
```

Comment:
minor, why named with lazy_ prefix, seems unnecessary

Reply:
Moving initialization logic based on Li Wei's suggestion below, so it will no longer be lazy. Will rename.